container-apigentools

```diff
@@ -30,7 +30,18 @@ if [[ "$1" == "--help" || "$1" == "-h" ]]; then
     exit 0
 fi
 
-IMAGE="$1"
+# if the user passed in something like apigentools/apigentools:latest, let's try using
+# a tag of that image that has for of "git-1234abc" - this will get saved to .apigentools-info
+# and is necessary for reproducibility and trackability - otherwise the user passed in
+# e.g. apigentools/apigentools:0.1.0 or apigentools/apigentools:git-1234abc already
+IMAGE=$1
+if echo "$1" | grep -q ":latest$"; then
+    IMAGE_WITH_GIT_TAG=`docker inspect $1 -f "{{range .RepoTags}}{{println .}}{{end}}" | grep ":git-.......$"`
```

Comment:
Will docker inspect get the latest if there are updates remotely, or.will it only see the latest you've manually pulled?

Reply:
Great point, it won't pull it in. I'm however reluctant to do an explicit pull on every single invocation of this script. How about we make this script behave like `docker run`, which only pulls if such an image is not present locally?

Reply:
Done, please re-review.

Reply:
Don't we want to pull the latest if there are updates the user doesn't have? Currently my understanding is that right now we'd only pull the actual latest if the user didn't have the image at all locally. Wouldn't we want to make sure they always have the latest latest?

Reply:
Well I think what we should do is whatever `docker run` does - which is "pull the image only if not present already". Pulling every time is IMO problematic, because it takes some time, so it's pretty annoying locally. It could also be problematic for people who have to limit internet traffic (like strict company policies for only working with stuff from intranet, working on a plane, ...)